client-v2/src/shared/bundles/collectionsBundle.ts

```diff
@@ -49,10 +57,65 @@ const collectionSelectors = {
   }
 };
 
+const SET_HIDDEN = 'SET_HIDDEN' as 'SET_HIDDEN';
+
+const setHidden = (collectionId: string, isHidden: boolean) => ({
+  entity: collectionsEntityName,
+  type: SET_HIDDEN,
+  payload: {
+    collectionId,
+    isHidden
+  }
+});
+
+const setHiddenAndPersist = addPersistMetaToAction(setHidden, {
+  persistTo: 'collection',
+  key: 'collectionId',
+  entity: 'collection'
+});
+
+export type SetHidden = ReturnType<typeof setHidden>;
+
+const collectionActions = {
+  ...actions,
+  setHiddenAndPersist
+};
+
+type CollectionActions = Actions<Collection> | SetHidden;
+
+const collectionReducer = (
+  state: State<Collection>,
+  action: CollectionActions
+): State<Collection> => {
+  const updatedState = reducer(state, action);
+  switch (action.type) {
+    case SET_HIDDEN: {
+      if (!updatedState.data[action.payload.collectionId]) {
+        return updatedState;
+      }
+
+      const freshState = {
```

Comment:
We could use 'lodash/fp/set' to save some lines here -- see e.g frontsReducer.ts.

Reply:
Done.